engine/access/state_stream/backend/handler.go

```diff
@@ -213,3 +215,155 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+// convertAccountsStatusesResultsToMessage converts account status responses to the message
+func convertAccountsStatusesResultsToMessage(
+	eventVersion entities.EventEncodingVersion,
+	resp *AccountStatusesResponse,
+) ([]*executiondata.SubscribeAccountStatusesResponse_Result, error) {
+	var results []*executiondata.SubscribeAccountStatusesResponse_Result
+	for address, events := range resp.AccountEvents {
+		convertedEvent, err := convert.EventsToMessagesWithEncodingConversion(events, entities.EventEncodingVersion_CCF_V0, eventVersion)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not convert events to entity: %v", err)
+		}
+
+		results = append(results, &executiondata.SubscribeAccountStatusesResponse_Result{
+			Address: flow.HexToAddress(address).Bytes(),
+			Events:  convertedEvent,
+		})
+	}
+	return results, nil
+}
+
+// sendSubscribeAccountStatusesResponseFunc defines the function signature for sending account status responses
+type sendSubscribeAccountStatusesResponseFunc func(*executiondata.SubscribeAccountStatusesResponse) error
+
+// handleAccountStatusesResponse handles account status responses by converting them to the message and sending them to the subscriber.
+func (h *Handler) handleAccountStatusesResponse(
+	requestHeartbeatInterval uint64,
+	evenVersion entities.EventEncodingVersion,
+	send sendSubscribeAccountStatusesResponseFunc,
+) func(resp *AccountStatusesResponse) error {
+	heartbeatInterval := requestHeartbeatInterval
+	if heartbeatInterval == 0 {
+		heartbeatInterval = h.defaultHeartbeatInterval
+	}
```

Comment:
nit: you can skip renaming the variable here.
```suggestion
func (h *Handler) handleAccountStatusesResponse(
	heartbeatInterval uint64,
	evenVersion entities.EventEncodingVersion,
	send sendSubscribeAccountStatusesResponseFunc,
) func(resp *AccountStatusesResponse) error {
	if heartbeatInterval == 0 {
		heartbeatInterval = h.defaultHeartbeatInterval
	}
```